flipperdevices/flipperzero-firmware
# heap: increased size

# What's new

- reduced reserved memory size for system stack
- +3.5k free RAM for system heap
- there's an existing MPU region protection for tail of system stack region - no additional checks required
- fbt: relink elf file on linker script changes

# Verification 

- run `free` in cli on dev and on this brach; compare total heap size
- run interrupt-heavy tasks 
- try reducing stack size in linker script even further (<=0x100) and observe a HardFault with `MMFSR->MSTKERR==1` 

> unstacking for an exception return has caused one or more access violations.
This fault is chained to the handler which means that the original return stack is still present.
The processor has not adjusted the SP from the failing return, and has not performed a new
save. The processor has not written a fault address to the MMFAR. Potential reasons:
a) Stack pointer is corrupted
b) MPU region for the stack changed during execution of the exception handler.

# Checklist (For Reviewer)
